libpod/image/tree.go

```diff
@@ -0,0 +1,138 @@
+package image
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/docker/go-units"
+	"github.com/pkg/errors"
+)
+
+const (
+	middleItem   = "├── "
+	continueItem = "│   "
+	lastItem     = "└── "
+)
+
+type tree struct {
+	img       *Image
+	imageInfo *InfoImage
+	layerInfo map[string]*LayerInfo
+	sb        *strings.Builder
+}
+
+// NewTree creates an image tree string representation for displaying it to the
+// user.
+func (i *Image) NewTree(whatRequires bool) (string, error) {
```

Comment:
I still don't like `NewTree` as it's not a constructor for a `Tree` object but it generates a string. I'd much prefer `Tree` or `GenerateTree`.

Reply:
Alright, renamed it to `GenerateTree`